src/hx/doctest/DocTestRunner.hx

```diff
@@ -207,41 +207,78 @@ interface DocTestResults {
 
     public function add(success:Bool, msg:String, loc:SourceLocation, pos:haxe.PosInfos):Void;
 
+    public function toString():String;
+
+    public function getTotalCount():Int;
     public function getSuccessCount():Int;
     public function getFailureCount():Int;
     public function logFailures():Void;
 }
 
 
 class DefaultDocTestResults implements DocTestResults {
-
-    var _testsOK = 0;
-    var _testsFailed = new Array<LogEvent>();
+    /**
+     * List of all tests sorted by succes
+     */
+    var _testsRunned = [
+        "total" => new Array<LogEvent>(),
+        "passed" => new Array<LogEvent>(),
+        "failed" => new Array<LogEvent>(),
+    ];
 
     inline
     public function new() {
     }
 
+    /**
+     * Logs and stores test results
+     * 
+     * @param success .
+     * @param msg message to print
+     * @param loc 
+     * @param pos 
+     */
     public function add(success:Bool, msg:String, loc:SourceLocation, pos:haxe.PosInfos) {
-        if(success) {
-            Logger.log(OK, msg, null, pos);
-            _testsOK++;
-        } else {
-            _testsFailed.push(Logger.log(ERROR, msg, loc, pos));
-        }
+        var event = new LogEvent(success ? OK : ERROR, msg, loc, pos);
+        event.log();
+
+        _testsRunned.get("total").push(event);
+        _testsRunned.get(success ? "passed" : "failed").push(event);
     }
 
     public function getSuccessCount():Int {
-        return _testsOK;
+        return _testsRunned.get("passed").length;
+    }
+
+    public function getTotalCount():Int {
+        return _testsRunned.get("total").length;
     }
 
     public function getFailureCount():Int {
-        return _testsFailed.length;
+        return _testsRunned.get("failed").length;
     }
 
+    /**
+     * Prints all errors in output
+     */
     public function logFailures():Void {
-        for (event in _testsFailed) {
+        var tests = _testsRunned.get("failed");
+        for (event in tests) {
             event.log(true);
         }
     }
+
+    /**
+     * Returns all results log as plain string
+     * @return String all runned tests log
+     */
+    public function toString():String {
+        var log = "";
+        var tests = _testsRunned.get("total");
+        for(event in tests) {
+            log += '\n' + event.toString(true);
```

Comment:
When you have a project with a lot of tests, this can result in an out of memory situation as you may be generating a really big string. The toString() should only return a summary, e.g. number of failed/passed tests.

Reply:
The whole point was to get all output. I don't see this as issue - even for utf-16 string with length around 200 symbols and for 10k test you will get 2b*200*10000=4000000b or 3906kb. 
Also i don't intend to test whole project in one time, it just pointless. But we can rename it in `generateFullLogStory` or something. Also you don't feel good about this code

Reply:
The whole point was to get all output. I don't see this as issue - even for utf-16 string with length around 200 symbols and for 10k test you will get 2b20010000=4000000b or 3906kb.
Also i don't intend to test whole project in one time, it just pointless. But we can rename it in `generateFullLogStory` or something. 
Also if you still don't like this function in your project we can just make public access to `_testRunned`